code/modules/projectiles/projectile/energy/tesla.dm

```diff
@@ -29,14 +29,12 @@
 	name = "tesla bolt"
 	icon_state = null
 	hitscan = TRUE
+	muzzle_type = /obj/effect/projectile/muzzle/stun
+	tracer_type = /obj/effect/projectile/tracer/stun
+	impact_type = /obj/effect/projectile/impact/stun
+	impact_effect_type = null
```

Comment:
Can't you just make a new tracer with random lighting icon_state, and no muzzle/impact?

Reply:
Aaaah sure fine I'll put in the extra effort later.

Reply:
tomorrow:tm:

Reply:
Starsector and insomnia has consumed my life right now I'm sorry.